src/cursor/cur_std.c

```diff
@@ -1290,6 +1290,46 @@ __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
     API_END_RET_STAT(session, ret, cursor_bound);
 }
 
+/*
+ * __wt_cursor_bounds_save --
+ *     Save cursor bounds to restore the state.
+ */
+void
+__wt_cursor_bounds_save(WT_CURSOR *cursor, WT_CURSOR_BOUNDS_STATE *state)
+{
+    WT_ITEM_SET(state->lower_bound, cursor->lower_bound);
+    WT_ITEM_SET(state->upper_bound, cursor->upper_bound);
+    /* Save the bound flags to the state. */
+    state->bound_flags = F_MASK(cursor, WT_CURSTD_BOUND_ALL);
+}
+
+/*
+ * __wt_cursor_bounds_restore --
+ *     Restore the cursor's bounds state. We want to change only related flags as we can't guarantee
+ *     the initial flag state of the primary and secondary column groups are the same.
```

Comment:
I would update this to have "column group cursors are the same".

Reply:
[23a8382](https://github.com/wiredtiger/wiredtiger/pull/8265/commits/23a8382b74954e1cbb8d5a7067b5eeb249828ebe)